From 184a45e62ad85986fcc7cbb4bfb50e82450687bf Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Fri, 30 Sep 2022 22:53:36 +0530
Subject: [PATCH] Makefile: Set VERSION & REVISION only if unset

This is useful while building nerdctl from git tag.
For example, while building rpm from a tagged source tarball.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 1085d00..d4362a1 100644
--- a/Makefile
+++ b/Makefile
@@ -27,9 +27,9 @@ endif
 PACKAGE := github.com/containerd/nerdctl
 BINDIR ?= /usr/local/bin
 
-VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
+VERSION ?= $(shell git describe --match 'v[0-9]*' --dirty='.m' --always --tags)
 VERSION_TRIMMED := $(VERSION:v%=%)
-REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+REVISION ?= $(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
 
 export GO_BUILD=GO111MODULE=on CGO_ENABLED=0 GOOS=$(GOOS) $(GO) build -ldflags "-s -w -X $(PACKAGE)/pkg/version.Version=$(VERSION) -X $(PACKAGE)/pkg/version.Revision=$(REVISION)"
 
-- 
2.30.5

